R/zzz.R

```diff
@@ -160,3 +160,83 @@ path_picker <- function(put_code, summary, pth_single) {
     file.path(pth_single, "summary")
   }
 }
+
+#'##extract BibTeX from ORCID string if available
+#'@param x the output of `cite_put()`
+#'
+#'@return Function returns a formated BibTeX record, if nothing
+#'was found or the BibTeX record is invalid, the input is passed
+#'through without modifying it
+#'@md
+#'@nord
+extract_BibTeX <- function(x) {
+  ##do we have a BibTeX entry? if not do nothing
+  if (grepl(pattern = "\"citation-type\" : \"bibtex\"",
+            x = x,
+            fixed = TRUE)) {
+
+
+    ##remove all line breaks, they cause too many problems
+    x <- gsub("\\n", replacement = "", x, fixed = TRUE, useBytes = TRUE)
+    x <- gsub("\n", replacement = "", x, fixed = TRUE, useBytes = TRUE)
+
+    ##this string extraction might be fragile ... on verra
+    bib <- substr(
+      x = x,
+      start = regexpr("@[a-z]+\\{", x, perl = TRUE, useBytes = FALSE),
+      stop =  regexpr(
+        "\\}[^a-z\\}]+\\}\\,[^a-z\\}]+(?=\\\"type\\\")",
+        x,
+        perl = TRUE,
+        useBytes = FALSE
+      )
+    )
+
+    ##make sure we do not break the function
+    ##and if we find no valid BibTeX entry we show where
+    if(bib == ""){
```

Comment:
use spaces in if and else statements, e.g., `if (` instead of `if(` and `) {` instead of `){` and `} else {` instead of `}else{`

Reply:
Done